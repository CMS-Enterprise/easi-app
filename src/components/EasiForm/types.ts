import React from 'react';
import {
  FieldPath,
  FieldValues,
  RefCallBack,
  RegisterOptions,
  UseFormRegisterReturn,
  UseFormReturn
} from 'react-hook-form';

export type EasiPartialSubmitProps<TFieldValues extends FieldValues> = {
  update: (
    /** Object containing valid dirty field values */
    formData: Partial<TFieldValues>
  ) => Promise<any>;
  /** Callback to execute after updating */
  callback?: () => void;
  /** Whether to clear field error messages */
  clearErrors?: boolean;
};

export type EasiPartialSubmit<TFieldValues extends FieldValues> = ({
  update,
  clearErrors,
  callback
}: EasiPartialSubmitProps<TFieldValues>) => Promise<void>;

export type EasiFormRegister<TFieldValues extends FieldValues> = <
  TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
>(
  name: TFieldName,
  options?: RegisterOptions<TFieldValues, TFieldName>
) => UseFormRegisterReturn<TFieldName> & { inputRef: RefCallBack };

export type UseEasiFormReturn<
  TFieldValues extends FieldValues = FieldValues,
  TContext = any
> = Omit<UseFormReturn<TFieldValues, TContext>, 'register'> & {
  /** Ignore errors and submit valid dirty field values */
  partialSubmit: EasiPartialSubmit<TFieldValues>;
  /** Custom register function that returns `inputRef` for function components */
  register: EasiFormRegister<TFieldValues>;
};

export type EasiFormProviderProps<
  TFieldValues extends FieldValues,
  TContext = any
> = {
  children: React.ReactNode | React.ReactNode[];
} & UseEasiFormReturn<TFieldValues, TContext>;
