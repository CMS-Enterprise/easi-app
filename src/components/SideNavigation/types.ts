export type SubNavLinkProps = {
  route: string;
  text: string;
};

export type NavLinkProps = SubNavLinkProps & {
  children?: SubNavLinkProps[];
  /** Designates end of navigation group with bottom border */
  groupEnd?: boolean;
};
