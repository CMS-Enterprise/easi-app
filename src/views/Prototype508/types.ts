import { DateTime } from 'luxon';

export enum DocumentType {
  TestPlan,
  TestingVPAT,
  TestResults,
  AwardedVPAT,
  RemediationPlan
}

export type Document = {
  type: DocumentType;
  mimetype: string;
  createdAt: DateTime;
};

export enum ProjectStatus {
  ConsultRequested,
  TestingRequested,
  TestingInProgress,
  TestingCompleted,
  InRemediation
}

export type BusinessOwner = {
  name: string;
};

export enum ActivityType {
  NoteAdded,
  StatusChanged,
  DocumentAdded,
  DocumentRemoved,
  ProjectCreated
}

export type Activity = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
  type: ActivityType;
};

export type Project = {
  id: number;
  name: string;
  status: ProjectStatus;
  businessOwner: BusinessOwner;
  submissionDate: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  activities: Activity[];
};
