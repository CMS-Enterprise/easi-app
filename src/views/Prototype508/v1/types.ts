import { DateTime } from 'luxon';

export enum DocumentType {
  TestPlan = 'Testing Plan',
  TestingVPAT = 'Testing VPAT',
  TestResults = 'Test Results',
  AwardedVPAT = 'Awarded VPAT',
  RemediationPlan = 'Remediation Plan'
}

export type Document = {
  id: number;
  type: DocumentType;
  mimetype: string;
  createdAt: DateTime;
  score?: number;
};

export enum ProjectStatus {
  ConsultRequested = 'Consult Requested',
  TestingRequested = 'Testing Requested',
  TestingInProgress = 'Testing in Progress',
  TestingCompleted = 'Testing Completed',
  InRemediation = 'In Remediation'
}

export type BusinessOwner = {
  name: string;
};

export enum ActivityType {
  NoteAdded,
  StatusChanged,
  DocumentAdded,
  DocumentRemoved,
  ProjectCreated
}

export type Activity = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
  type: ActivityType;
};

export type Project = {
  id: number;
  name: string;
  status: ProjectStatus;
  businessOwner: BusinessOwner;
  submissionDate: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  activities: Activity[];
  banner?: string;
};
