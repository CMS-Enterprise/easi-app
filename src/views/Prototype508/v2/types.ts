import { DateTime } from 'luxon';

import { ProgressStatus } from './components/Progress';

export enum DocumentType {
  TestPlan = 'Testing Plan',
  TestingVPAT = 'Testing VPAT',
  TestResults = 'Test Results',
  AwardedVPAT = 'Awarded VPAT',
  RemediationPlan = 'Remediation Plan',
  Other = 'Other Document'
}

export type Document = {
  id: number;
  type: DocumentType;
  otherName?: string;
  mimetype: string;
  createdAt: DateTime;
  testDate?: DateTime;
  score?: number;
};

export type Person = {
  name: string;
  component: string;
};

export type Note = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
};

export enum RequestStep {
  RequestReceived = 'Request Received',
  DocumentsReceived = 'Documents Received',
  TestScheduled = 'Test Scheduled',
  RemediationRequested = 'Remediation Requested',
  RemediationInProgress = 'Remediation in Progress',
  ValidationTestingScheduled = 'Validation Testing Scheduled',
  Completed = 'Completed'
}

export type RequestStepStatus = {
  date: DateTime;
  status: ProgressStatus;
};

export type PastRequest = {
  name: string;
  release: string;
  lastTestDate: DateTime;
};

export type Project = {
  id: number;
  name: string;
  release?: string;
  status: RequestStep;
  businessOwner: Person;
  pointOfContact: Person;
  testDate?: DateTime;
  submissionDate: DateTime;
  remediationStartDate?: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  notes: Note[];
  banner?: string;
  stepStatuses: { [index in RequestStep]?: RequestStepStatus };
  pastRequests: PastRequest[];
};
