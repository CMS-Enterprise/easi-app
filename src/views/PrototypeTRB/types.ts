import { DateTime } from 'luxon';

export enum DocumentType {
  FindingsLetter = 'Findings Letter',
  ArchitectureDiagram = 'Architecture Diagram',
  Deck = 'Deck'
}

export type Document = {
  id: number;
  type: DocumentType;
  mimetype: string;
  createdAt: DateTime;
  score?: number;
};

export enum ProjectStatus {
  ReviewRequested = 'Review Requested',
  InConsultation = 'In Consultation',
  InDesignReview = 'In Design Review',
  InOperationalReadiness = 'In Operational Readiness',
  WaitingForFindings = 'Waiting for Findings',
  Closed = 'Closed'
}

export type BusinessOwner = {
  name: string;
};

export enum ActivityType {
  NoteAdded,
  StatusChanged,
  DocumentAdded,
  DocumentRemoved,
  ProjectCreated
}

export type Activity = {
  id: number;
  authorName: string;
  content: string;
  createdAt: DateTime;
  type: ActivityType;
};

export type Project = {
  id: number;
  name: string;
  status: ProjectStatus;
  businessOwner: BusinessOwner;
  submissionDate: DateTime;
  lastUpdatedAt: DateTime;
  lifecycleID: string;
  description?: string;
  documents: Document[];
  activities: Activity[];
  banner?: string;
};
