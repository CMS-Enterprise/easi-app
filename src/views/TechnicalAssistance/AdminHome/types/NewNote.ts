/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TRBAdminNoteCategory } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL fragment: NewNote
// ====================================================

export interface NewNote_author {
  __typename: "UserInfo";
  commonName: string;
}

export interface NewNote {
  __typename: "TRBAdminNote";
  id: UUID;
  isArchived: boolean;
  category: TRBAdminNoteCategory;
  noteText: string;
  author: NewNote_author;
  createdAt: Time;
}
