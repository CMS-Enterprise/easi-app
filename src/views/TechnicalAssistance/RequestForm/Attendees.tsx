import React, { useCallback, useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import {
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch
} from 'react-router-dom';
import { ApolloError, ApolloQueryResult } from '@apollo/client';
import { yupResolver } from '@hookform/resolvers/yup';
import { Form, IconArrowBack } from '@trussworks/react-uswds';
import classNames from 'classnames';

import UswdsReactLink from 'components/LinkWrapper';
import PageLoading from 'components/PageLoading';
import Alert from 'components/shared/Alert';
import Divider from 'components/shared/Divider';
import useTRBAttendees from 'hooks/useTRBAttendees';
import {
  GetTrbRequest,
  GetTrbRequest_trbRequest as TrbRequest,
  GetTrbRequestVariables
} from 'queries/types/GetTrbRequest';
import { TRBAttendee } from 'queries/types/TRBAttendee';
import { PersonRole } from 'types/graphql-global-types';
import {
  AttendeeFieldLabels,
  TRBAttendeeFields
} from 'types/technicalAssistance';
import { trbRequesterSchema } from 'validations/trbRequestSchema';

import { AttendeeFields, AttendeesTable } from './AttendeesForm/components';
import AttendeesForm from './AttendeesForm';
import Pager from './Pager';
import { FormStepComponentProps, StepSubmit, TrbFormAlert } from '.';

// Make FormStepComponentProps conditionally required on the presence of fromTaskList
// Used to render Attendees/form from task list outside the scope of initial request form
type AttendeesProps =
  | {
      fromTaskList: true;
      request?: TrbRequest;
      /** Refetch the trb request from the form wrapper */
      refetchRequest?: (
        variables?: Partial<GetTrbRequestVariables> | undefined
      ) => Promise<ApolloQueryResult<GetTrbRequest>>;
      /**
       * Set the current form step component submit handler
       * so that in can be used in other places like the header.
       * Form step components need to reassign the handler.
       */
      setStepSubmit?: React.Dispatch<React.SetStateAction<StepSubmit | null>>;
      /** Set to update the submitting state from step components to the parent request form */
      setIsStepSubmitting?: React.Dispatch<React.SetStateAction<boolean>>;
      /** Set a form level alert message from within step components */
      setFormAlert: React.Dispatch<React.SetStateAction<TrbFormAlert>>;
      stepUrl?: {
        current: string;
        next: string;
        back: string;
      };
      taskListUrl: string;
    }
  | ({ fromTaskList?: false } & FormStepComponentProps);

/** Initial blank attendee object */
export const initialAttendee: TRBAttendee = {
  __typename: 'TRBRequestAttendee',
  trbRequestId: '',
  id: '',
  userInfo: {
    __typename: 'UserInfo',
    commonName: '',
    euaUserId: '',
    email: ''
  },
  component: null,
  role: null,
  createdAt: ''
};

function Attendees({
  request,
  stepUrl,
  setFormAlert,
  refetchRequest,
  setIsStepSubmitting,
  setStepSubmit,
  taskListUrl,
  fromTaskList
}: AttendeesProps) {
  const { t } = useTranslation('technicalAssistance');
  const { path, url } = useRouteMatch();
  const history = useHistory();

  const { id: trbID } = useParams<{
    id: string;
  }>();

  const taskListAttendeesURL = `/trb/task-list/${trbID}/attendees`;

  /** Field labels object from translation file */
  const fieldLabels: AttendeeFieldLabels = t(
    'attendees.fieldLabels.requester',
    {
      returnObjects: true
    }
  );

  /**
   * Active attendee for form fields
   *
   * Used to set field values when creating or editing attendee
   */
  const [activeAttendee, setActiveAttendee] = useState<TRBAttendee>({
    ...initialAttendee,
    trbRequestId: request?.id || trbID
  });

  // Initialize form
  const {
    control,
    handleSubmit,
    setValue,
    clearErrors,
    getValues,
    reset,
    formState: { errors, isSubmitting, isDirty }
  } = useForm<TRBAttendeeFields>({
    resolver: yupResolver(trbRequesterSchema)
  });

  /**
   * Get TRB attendees data and mutations
   */
  const {
    data: { attendees, requester, loading },
    updateAttendee,
    deleteAttendee
  } = useTRBAttendees(request?.id || trbID);

  /**
   * Reset form with default values after useTRBAttendees query returns requester
   */
  useEffect(() => {
    /** Default reqiester values */
    const defaultValues: TRBAttendeeFields = {
      euaUserId: requester?.userInfo?.euaUserId || '',
      component: requester?.component,
      role: requester?.role
    };
    // Reset form
    reset(defaultValues);
  }, [requester, reset]);

  /** Submit requester as attendee */
  const submitForm = useCallback<StepSubmit>(
    callback =>
      // Start the submit promise
      handleSubmit(
        // Validation passed
        async formData => {
          try {
            // Submit the input only if there are changes
            if (isDirty && requester.id) {
              const { component, role } = formData;
              // Update requester
              await updateAttendee({
                id: requester.id,
                component: component || '',
                role: role as PersonRole
              })
                // Refresh the RequestForm parent request query
                // to update things like `stepsCompleted`
                .then(() => refetchRequest?.());
            }
            callback?.();
          } catch (e) {
            if (e instanceof ApolloError) {
              setFormAlert({
                type: 'error',
                message: t<string>('attendees.alerts.error')
              });
            }
          }
        }
      )(),
    [
      handleSubmit,
      isDirty,
      refetchRequest,
      requester,
      updateAttendee,
      setFormAlert,
      t
    ]
  );

  useEffect(() => {
    if (setStepSubmit) setStepSubmit(() => submitForm);
  }, [setStepSubmit, submitForm]);

  useEffect(() => {
    if (setIsStepSubmitting) setIsStepSubmitting(isSubmitting);
  }, [setIsStepSubmitting, isSubmitting]);

  // Wait until attendees query has completed to load form
  if (loading) return <PageLoading />;

  if (fromTaskList && attendees.length === 0 && !path.includes('list')) {
    history.push(`${taskListAttendeesURL}/list`);
  }

  return (
    <div
      className={classNames('trb-attendees', {
        'margin-top-2': !fromTaskList
      })}
    >
      <Switch>
        <Route exact path={`${path}/list`}>
          <AttendeesForm
            backToFormUrl={stepUrl?.current || taskListAttendeesURL}
            activeAttendee={activeAttendee}
            setActiveAttendee={setActiveAttendee}
            trbRequestId={request?.id || trbID}
            setFormAlert={setFormAlert}
            taskListUrl={taskListUrl}
            fromTaskList={fromTaskList}
          />
        </Route>

        <Route exact path={`${path}`}>
          <Form
            data-testid="trb-attendees-form"
            className="margin-bottom-4 maxw-full"
            onSubmit={e => e.preventDefault()}
          >
            {/* Requester Fields */}
            {!fromTaskList && (
              <>
                <AttendeeFields
                  type="requester"
                  activeAttendee={requester}
                  fieldLabels={fieldLabels}
                  errors={errors}
                  clearErrors={clearErrors}
                  setValue={setValue}
                  control={control}
                />

                <Divider className="margin-top-4" />
              </>
            )}

            {fromTaskList && (
              <>
                <h1 className="margin-bottom-0 margin-top-4">
                  {t('attendees.heading')}
                </h1>
                <p className="line-height-body-5 margin-top-0 margin-bottom-2">
                  {t('attendees.description')}
                </p>

                <UswdsReactLink
                  to={`/trb/task-list/${trbID}`}
                  className="display-block margin-bottom-5"
                >
                  <IconArrowBack className="margin-right-1 text-middle" />
                  <span className="line-height-body-5">
                    {t('requestFeedback.returnToTaskList')}
                  </span>
                </UswdsReactLink>

                {attendees?.length === 0 && (
                  <Alert type="info" slim className="margin-bottom-3">
                    {t('attendees.noAttendees')}
                  </Alert>
                )}
              </>
            )}

            <div className="margin-bottom-6">
              {!fromTaskList && (
                <h4 className="margin-bottom-2">
                  {t('attendees.additionalAttendees')}
                </h4>
              )}

              {/* Button to add additional attendee */}
              <UswdsReactLink
                variant="unstyled"
                className={classNames('usa-button', 'margin-top-0', {
                  'usa-button--outline': attendees.length > 0
                })}
                to={`${url}/list`}
                // Save requester when navigating to additional attendees form
                onClick={() => {
                  // Clear out alert from previous action
                  setFormAlert(false);

                  // Check that all requester fields are filled out
                  // This submission should not throw errors if missing fields
                  if (
                    getValues().euaUserId &&
                    getValues().component &&
                    getValues().role
                  ) {
                    submitForm(() => history.push(`${url}/list`));
                  }
                }}
              >
                {t(
                  attendees.length > 0
                    ? 'attendees.addAnotherAttendee'
                    : 'attendees.addAnAttendee'
                )}
              </UswdsReactLink>

              {/* List of additional attendees */}
              <AttendeesTable
                attendees={attendees}
                setActiveAttendee={setActiveAttendee}
                setFormAlert={setFormAlert}
                deleteAttendee={(id: string) => deleteAttendee(id)}
                trbRequestId={request?.id || trbID}
              />
            </div>

            <Pager
              className="margin-top-5"
              next={
                // Hides button on task list
                stepUrl && {
                  disabled: isSubmitting,
                  onClick: () => {
                    submitForm(() => {
                      history.push(stepUrl.next);
                    });
                  },
                  text: t(
                    attendees.length > 0
                      ? 'Next'
                      : 'attendees.continueWithoutAdding'
                  ),
                  outline: attendees.length === 0
                }
              }
              back={
                stepUrl && {
                  disabled: isSubmitting,
                  onClick: () => {
                    submitForm(() => {
                      history.push(stepUrl.back);
                    });
                  }
                }
              }
              saveExitDisabled={isSubmitting}
              submit={submitForm}
              submitDisabled={!stepUrl}
              taskListUrl={taskListUrl}
            />
          </Form>
        </Route>
      </Switch>
    </div>
  );
}

export default Attendees;
