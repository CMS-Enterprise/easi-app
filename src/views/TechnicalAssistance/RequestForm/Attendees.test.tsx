import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ApolloQueryResult, NetworkStatus } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { attendees, requester, trbRequest } from 'data/mock/trbRequest';
import GetCedarContactsQuery from 'queries/GetCedarContactsQuery';
import GetTrbRequestQuery from 'queries/GetTrbRequestQuery';
import { GetTRBRequestAttendeesQuery } from 'queries/TrbAttendeeQueries';
import { GetTrbRequest } from 'queries/types/GetTrbRequest';
import { TRBAttendee } from 'queries/types/TRBAttendee';

import Attendees from './Attendees';

/** Requester object with initial null values for component and role */
const initialRequester: TRBAttendee = {
  ...requester,
  component: null,
  role: null
};

describe('Trb Request form: Attendees', () => {
  const mockRefetch = async (): Promise<ApolloQueryResult<GetTrbRequest>> => {
    return {
      loading: false,
      networkStatus: NetworkStatus.ready,
      data: {
        trbRequest
      }
    };
  };

  const getAttendeesQuery = {
    request: {
      query: GetTRBRequestAttendeesQuery,
      variables: {
        id: trbRequest.id
      }
    },
    result: {
      data: {
        trbRequest: {
          id: trbRequest.id,
          attendees: [initialRequester, ...attendees]
        }
      }
    }
  };

  const getTrbRequestQuery = {
    request: {
      query: GetTrbRequestQuery,
      variables: {
        id: trbRequest.id
      }
    },
    result: {
      data: trbRequest
    }
  };

  const getCedarContactsQuery = {
    request: {
      query: GetCedarContactsQuery,
      variables: {
        commonName: initialRequester.userInfo?.commonName
      }
    },
    result: {
      data: requester
    }
  };

  it('Renders the attendees form', async () => {
    const { getByTestId, findByTestId } = render(
      <MemoryRouter>
        <MockedProvider
          mocks={[getAttendeesQuery, getTrbRequestQuery, getCedarContactsQuery]}
          addTypename={false}
        >
          <Attendees
            request={trbRequest}
            stepUrl={{
              current: `/trb/requests/${trbRequest.id}/attendees`,
              next: `/trb/requests/${trbRequest.id}/documents`,
              back: `/trb/requests/${trbRequest.id}/subject`
            }}
            refetchRequest={mockRefetch}
            setIsStepSubmitting={() => {}}
            setStepSubmit={() => {}}
            setFormAlert={() => {}}
            taskListUrl=""
          />
        </MockedProvider>
      </MemoryRouter>
    );

    const requesterContactSelect = await findByTestId('cedar-contact-select');

    // Check that requester name and EUA matches mocked query results
    expect(requesterContactSelect).toHaveValue(
      `${requester.userInfo?.commonName}, ${requester.userInfo?.euaUserId}`
    );

    const requesterComponentField = getByTestId('trb-requester-component');
    const requesterRoleField = getByTestId('trb-requester-role');

    // Check that form loads initial null values for requester component and role
    expect(requesterComponentField).not.toHaveValue();
    expect(requesterRoleField).not.toHaveValue();

    // Select requester component
    userEvent.selectOptions(requesterComponentField, [requester.component!]);
    expect(requesterComponentField).toHaveValue(requester.component);

    // Select requester role
    userEvent.selectOptions(requesterRoleField, [requester.role!]);
    expect(requesterRoleField).toHaveValue(requester.role);
  });
});
