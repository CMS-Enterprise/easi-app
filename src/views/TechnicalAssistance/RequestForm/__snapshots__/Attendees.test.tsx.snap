// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trb Request form: Attendees Renders the attendees form 1`] = `
<DocumentFragment>
  <div
    class="trb-attendees margin-top-2"
  >
    <form
      class="usa-form margin-bottom-4 maxw-full"
      data-testid="trb-attendees-form"
    >
      <div
        class="grid-row margin-bottom-5"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col-12 desktop:grid-col-6"
          data-testid="grid"
        >
          <div
            class="usa-form-group"
            data-testid="formGroup"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="euaUserId"
            >
              Requester
            </label>
            <div
              class="margin-top-1 cedar-contact-select usa-combo-box opacity-70 maxw-none cedar-contact-select--is-disabled css-3iigni-container"
              id="euaUserId"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-euaUserId-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class="cedar-contact-select__control cedar-contact-select__control--is-disabled css-7n4c1b-control"
              >
                <div
                  class="cedar-contact-select__value-container cedar-contact-select__value-container--has-value css-319lph-ValueContainer"
                >
                  <div
                    class="cedar-contact-select__input-container css-1pndypt-Input"
                    data-value="Jerry Seinfeld, SF13"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-label="Cedar-Users"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class="cedar-contact-select__input"
                      data-testid="cedar-contact-select"
                      disabled=""
                      id="react-select-euaUserId-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value="Jerry Seinfeld, SF13"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="usa-form-group margin-top-3"
            data-testid="formGroup"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="component"
            >
              Requester component
            </label>
            <select
              class="usa-select"
              data-testid="trb-requester-component"
              id="component"
              name="component"
            >
              <option
                disabled=""
                label="- Select -"
              />
              <option
                value="Center for Clinical Standards and Quality"
              >
                Center for Clinical Standards and Quality (CCSQ)
              </option>
              <option
                value="Center for Consumer Information and Insurance Oversight"
              >
                Center for Consumer Information and Insurance Oversight (CCIIO)
              </option>
              <option
                value="Center for Medicare"
              >
                Center for Medicare (CM)
              </option>
              <option
                value="Center for Medicaid and CHIP Services"
              >
                Center for Medicaid and CHIP Services (CMCS)
              </option>
              <option
                value="Center for Medicare and Medicaid Innovation"
              >
                Center for Medicare and Medicaid Innovation (CMMI)
              </option>
              <option
                value="Center for Program Integrity"
              >
                Center for Program Integrity (CPI)
              </option>
              <option
                value="CMS Wide"
              >
                CMS Wide (CMS)
              </option>
              <option
                value="Emergency Preparedness and Response Operations"
              >
                Emergency Preparedness and Response Operations (EPRO)
              </option>
              <option
                value="Federal Coordinated Health Care Office"
              >
                Federal Coordinated Health Care Office (FCHCO)
              </option>
              <option
                value="Office of Acquisition and Grants Management"
              >
                Office of Acquisition and Grants Management (OAGM)
              </option>
              <option
                value="Office of Burden Reduction and Health Informatics"
              >
                Office of Burden Reduction and Health Informatics (OBRHI)
              </option>
              <option
                value="Office of Communications"
              >
                Office of Communications (OC)
              </option>
              <option
                value="Office of Enterprise Data and Analytics"
              >
                Office of Enterprise Data and Analytics (OEDA)
              </option>
              <option
                value="Office of Equal Opportunity and Civil Rights"
              >
                Office of Equal Opportunity and Civil Rights (OEOCR)
              </option>
              <option
                value="Office of Financial Management"
              >
                Office of Financial Management (OFM)
              </option>
              <option
                value="Office of Human Capital"
              >
                Office of Human Capital (OHC)
              </option>
              <option
                value="Office of Information Technology"
              >
                Office of Information Technology (OIT)
              </option>
              <option
                value="Office of Legislation"
              >
                Office of Legislation (OL)
              </option>
              <option
                value="Office of Minority Health"
              >
                Office of Minority Health (OMH)
              </option>
              <option
                value="Office of Program Operations and Local Engagement"
              >
                Office of Program Operations and Local Engagement (OPOLE)
              </option>
              <option
                value="Office of Security, Facilities, and Logistics Operations"
              >
                Office of Security, Facilities, and Logistics Operations (OSFLO)
              </option>
              <option
                value="Office of Strategic Operations and Regulatory Affairs"
              >
                Office of Strategic Operations and Regulatory Affairs (OSORA)
              </option>
              <option
                value="Office of Strategy, Performance, and Results"
              >
                Office of Strategy, Performance, and Results (OSPR)
              </option>
              <option
                value="Office of the Actuary"
              >
                Office of the Actuary (OA)
              </option>
              <option
                value="Offices of Hearings and Inquiries"
              >
                Offices of Hearings and Inquiries (OHI)
              </option>
              <option
                value="Other"
              >
                Other
              </option>
            </select>
          </div>
          <div
            class="usa-form-group margin-top-3"
            data-testid="formGroup"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="role"
            >
              Requester role
            </label>
            <select
              class="usa-select"
              data-testid="trb-requester-role"
              id="role"
              name="role"
            >
              <option
                disabled=""
                label="- Select -"
              />
              <option
                label="Product Owner"
                value="PRODUCT_OWNER"
              />
              <option
                label="System Owner"
                value="SYSTEM_OWNER"
              />
              <option
                label="System Maintainer"
                value="SYSTEM_MAINTAINER"
              />
              <option
                label="Contracting Officer's Representative (COR)"
                value="CONTRACT_OFFICE_RSREPRESENTATIVE"
              />
              <option
                label="Cloud Navigator"
                value="CLOUD_NAVIGATOR"
              />
              <option
                label="Privacy Advisor"
                value="PRIVACY_ADVISOR"
              />
              <option
                label="CRA"
                value="CRA"
              />
              <option
                label="Other"
                value="OTHER"
              />
              <option
                label="Unknown"
                value="UNKNOWN"
              />
            </select>
          </div>
        </div>
      </div>
      <div
        class="easi-divider"
      />
      <div
        class="margin-bottom-6"
      >
        <h4
          class="margin-bottom-2"
        >
          Additional attendees
        </h4>
        <a
          class="usa-button margin-top-0 usa-button--outline"
          href="//list"
        >
          Add another attendee
        </a>
        <div
          class="trbAttendees-table margin-top-4 margin-bottom-neg-1"
        >
          <table
            class="usa-table usa-table--borderless src-components-Table-Table-module__fullwidth--fOyj6"
            data-testid="table"
          >
            <tbody
              class="grid-row grid-gap-sm"
              role="rowgroup"
            >
              <tr
                class="tablet:grid-col-6 margin-bottom-1"
                role="row"
              >
                <td
                  class="trbAttendee display-flex"
                  data-testid="trbAttendee-JTTC"
                  id="trbAttendee-JTTC"
                  role="cell"
                  style="padding: 0px;"
                >
                  <div
                    class="easi-initials-icon bg-accent-cool-lighter margin-right-1"
                  >
                    <span>
                      AJ
                    </span>
                  </div>
                  <div>
                    <p
                      class="margin-y-05 text-bold"
                    >
                      Anabelle Jerde, CPI
                    </p>
                    <p
                      class="margin-y-05"
                    >
                      anabelle.jerde@local.fake
                    </p>
                    <p
                      class="margin-top-05 margin-bottom-0"
                    >
                      Product Owner
                    </p>
                    <ul
                      class="usa-button-group margin-y-0"
                    >
                      <li
                        class="usa-button-group__item"
                      >
                        <a
                          class="margin-right-05"
                          href="//list"
                        >
                          Edit
                        </a>
                      </li>
                      <li
                        class="usa-button-group__item"
                      >
                        <button
                          class="usa-button usa-button--unstyled text-error margin-top-0"
                          data-testid="button"
                          type="button"
                        >
                          Remove
                        </button>
                      </li>
                    </ul>
                  </div>
                </td>
              </tr>
              <tr
                class="tablet:grid-col-6 margin-bottom-1"
                role="row"
              >
                <td
                  class="trbAttendee display-flex"
                  data-testid="trbAttendee-ADMI"
                  id="trbAttendee-ADMI"
                  role="cell"
                  style="padding: 0px;"
                >
                  <div
                    class="easi-initials-icon bg-accent-cool-lighter margin-right-1"
                  >
                    <span>
                      AA
                    </span>
                  </div>
                  <div>
                    <p
                      class="margin-y-05 text-bold"
                    >
                      Audrey Abrams, CMS
                    </p>
                    <p
                      class="margin-y-05"
                    >
                      audrey.abrams@local.fake
                    </p>
                    <p
                      class="margin-top-05 margin-bottom-0"
                    >
                      System Owner
                    </p>
                    <ul
                      class="usa-button-group margin-y-0"
                    >
                      <li
                        class="usa-button-group__item"
                      >
                        <a
                          class="margin-right-05"
                          href="//list"
                        >
                          Edit
                        </a>
                      </li>
                      <li
                        class="usa-button-group__item"
                      >
                        <button
                          class="usa-button usa-button--unstyled text-error margin-top-0"
                          data-testid="button"
                          type="button"
                        >
                          Remove
                        </button>
                      </li>
                    </ul>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="border-base-light border-top-1px margin-top-5"
      >
        <div
          class="margin-top-2"
        >
          <button
            class="usa-button usa-button--outline margin-top-0 margin-bottom-1 mobile-lg:margin-bottom-0"
            data-testid="button"
            type="button"
          >
            Back
          </button>
          <button
            class="usa-button margin-top-0"
            data-testid="button"
            type="submit"
          >
            Next
          </button>
        </div>
        <button
          class="usa-button usa-button--unstyled margin-top-2"
          data-testid="button"
          type="button"
        >
          <svg
            class="usa-icon margin-right-05 margin-bottom-2px text-tbottom"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Save and exit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
